superset-frontend/src/configs.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export function configs() {
+  // read superset config using multiple sources
+  // currently it only reads from bootstrap data
+  return configsFromBootstrapData();
+}
+
+function configsFromBootstrapData() {
```

Comment:
We generally get these configurations from Redux to only parse `data-bootstrap` once and also to have the ability to dynamically change the original settings and allow components to react accordingly.

Reply:
Is there any example. I have found several but they all read from bootstrap data https://sourcegraph.com/search?q=getAttribute%28%27data-bootstrap%27%29+repo%3Aapache%2Fsuperset%24

Reply:
One example I used for reference was https://github.com/apache/superset/blob/3311128c5e6c5de2ea5d6a2dfeb01ea3179e9af8/superset-frontend/src/views/store.ts#L65 It does have some redux related code but it still reads data-bootstrap

Reply:
Your reference is correct. In this file, the `common` state will be initialized in the Redux store with contents from `data-bootstrap`. This means that you don't need to parse the content again, you can access it using Redux like [this](https://github.com/apache/superset/blob/1388f21ee34251b6ef83beb009ba0901e4067848/superset-frontend/src/dashboard/containers/Dashboard.ts#L53). If any component dynamically changes the value of that configuration, your component would be automatically updated.

Reply:
I see. So if I want to read bootstrap data in this RefreshIntervalModal which belongs to Dashboard, does that mean this file should contain some logic to save bootstrapData? https://sourcegraph.com/github.com/apache/superset/-/blob/superset-frontend/src/dashboard/components/Dashboard.jsx In the file you linked in the previous comment, it reads from dashboardInfo but I don't see Dashboard.jsx saves the data. I'm wondering if I miss anything

Reply:
I think there was an error with the previous link. I update it to point to https://github.com/apache/superset/blob/1388f21ee34251b6ef83beb009ba0901e4067848/superset-frontend/src/SqlLab/components/TabbedSqlEditors/index.jsx#L352

> So if I want to read bootstrap data in this RefreshIntervalModal which belongs to Dashboard, does that mean this file should contain some logic to save bootstrapData?

You don't need to save `bootstrapData`, you just need to retrieve the values from Redux instead of parsing `data-bootstrap` again like you're doing in https://github.com/apache/superset/pull/21924/files#diff-9cbbe9c430f0761d26c381b58358e44e5c4f0ac353f223832e9150d4070413c4

Reply:
If you inspect the Redux state when you're inside any dashboard you'll see that the configuration values are already there.

<img width="1785" alt="Screen Shot 2022-10-25 at 3 03 42 PM" src="https://user-images.githubusercontent.com/70410625/197848619-ddd6969d-9a8b-4b5b-9f01-f3444b587a41.png">

Reply:
I see. I did a bit more research and I think I know how to retrieve the values. I feel I didn't make myself clear in the previous message as well. Actually I was wondering how those values were saved in redux at the first place.

Reply:
Updated